lib/fs.js

```diff
@@ -395,12 +395,32 @@ function readFileAfterClose(err) {
   else
     buffer = context.buffer;
 
-  if (context.encoding)
-    buffer = buffer.toString(context.encoding);
+  if (err) return callback(err, buffer);
 
-  callback(err, buffer);
+  if (context.encoding) {
+    const result = tryToString(buffer, context.encoding);
+    if (result.error)
+      return callback(result.error, buffer);
```

Comment:
Given that it removes a throw, it's at least a semver-minor or major, which means it wouldn't make it in to `v4.x`